src/main/java/hudson/remoting/RemoteClassLoader.java

```diff
@@ -794,6 +794,7 @@ public ClassLoaderProxy(@Nonnull ClassLoader cl, Channel channel) {
             this.channel = channel;
         }
 
+        @SuppressFBWarnings(value = "URLCONNECTION_SSRF_FD", justification = "This is only used for managing the jar cache as files.")
```

Comment:
Just because that's how it's supposed to be used doesn't mean that's what this method does. It's public, and it's exploitable by using non-file URLs.

Reply:
That's true. This really should have been implemented differently, but it's complicated to change that implementation at this point. Might be worth creating another PR at some point to clean some of this up. I don't know if there's a better way to phrase the suppression or to fix it at this point.